Modules/_sqlite/cursor.c

```diff
@@ -434,8 +434,14 @@ begin_transaction(pysqlite_Connection *self)
 
     Py_BEGIN_ALLOW_THREADS
     sqlite3_stmt *statement;
-    rc = sqlite3_prepare_v2(self->db, self->begin_statement, -1, &statement,
-                            NULL);
+    if (self->isolation_level != NULL) {
+        char begin_stmt[80] = {0, };
```

Comment:
IMO, 80 chars is too much overshoot. Max `strlen` is 15.

```suggestion
        char begin_stmt[16] = {0, };
        assert((strlen(self->isolation_level) + 1) >
               (sizeof(begin_stmt) - strlen(self->isolation_level)));
```